docs: moved local docker setup into local setup file (#35975)
freeCodeCamp/freeCodeCamp: docs/how-to-setup-freecodecamp-locally.md

# Set up freeCodeCamp locally
Follow these guidelines for getting freeCodeCamp locally on your system. This is highly recommended if you want to contribute regularly.

Some of the contribution workflows, like previewing pages for the guide or the coding challenges, debugging and fixing bugs in codebase, requires you to have freeCodeCamp running locally.

> The new Docker setup procedure *should* result in fewer, if not zero, errors during the setup process and make it easier for you. We use Docker to install and run many of the software dependencies which means that you don't have to install services like MongoDB or MailHog on your computer or worry about ensuring you are using the correct version.

Reviewer: Remove here, move until the users reach the actual setup.

[Skip running freeCodeCamp locally](#making-changes-to-your-clone-of-freecodecamp-locally)

There are currently two methods to run freeCodeCamp locally:
- Docker (recommended)
- Local

Only **one** of these methods is necessary. If you are having issues using one method, try using the other. If you do run into issues with either method, first perform a web search for your issue and see if it has already been answered. If you cannot find a solution, please search our GitHub [issues](https://github.com/freeCodeCamp/freeCodeCamp/issues) page for a solution and report the issue if it has not yet been reported.

### Installing prerequisites
Start by installing the prerequisite software:

Software required for both Docker and Local builds:

| Prerequisite | Version | Notes |
| ------------ | ------- | ----- |
| [Node.js](http://nodejs.org)| `10.x`  | [LTS Schedule](https://github.com/nodejs/Release#release-schedule) |
| npm (comes bundled with Node)| `6.x`   | Does not have LTS releases, we use the version bundled with Node LTS |

**Docker Build additional prerequisite:**

> | Prerequisite | Version | Notes |

Reviewer: ```suggestion

| Prerequisite | Version | Notes |
```

| ------------ | ------- | ----- |

> | [Docker CE](https://docs.docker.com/install/) | `Stable` |       |

Reviewer: ```suggestion
| [Docker CE](https://docs.docker.com/install/) | `Stable` | - |
```

**Local Build additional prerequisite:**

> | Prerequisite | Version | Notes |

Reviewer: ```suggestion

| Prerequisite | Version | Notes |
```

[Skip running freeCodeCamp locally](#making-changes-to-your-clone-of-freecodecamp-locally)

There are currently two methods to run freeCodeCamp locally:
- Docker (recommended)
- Local

> Only **one** of these methods is necessary. If you are having issues using one method, try using the other. If you do run into issues with either method, first perform a web search for your issue and see if it has already been answered. If you cannot find a solution, please search our GitHub [issues](https://github.com/freeCodeCamp/freeCodeCamp/issues) page for a solution and report the issue if it has not yet been reported.

Reviewer: ```suggestion
You are required to follow any **one** of the above methods. 

The Docker setup will ideally result in fewer errors during the installation process and aims to have the best developer experience. We use Docker to install and run the additional software dependencies behind the scenes. This should make it hassle free and consistent experience across most device/OS types.

If you are having issues using one method, try using the other. If you do run into issues with either method, first perform a web search for your issue and see if it has already been answered. If you cannot find a solution, please search our GitHub [issues](https://github.com/freeCodeCamp/freeCodeCamp/issues) page for a solution and report the issue if it has not yet been reported.

And as always, feel free to hop on to our [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors), for quick queries.
```

### Installing prerequisites
Start by installing the prerequisite software:

Software required for both Docker and Local builds:

| Prerequisite | Version | Notes |
| ------------ | ------- | ----- |
| [Node.js](http://nodejs.org)| `10.x`  | [LTS Schedule](https://github.com/nodejs/Release#release-schedule) |
| npm (comes bundled with Node)| `6.x`   | Does not have LTS releases, we use the version bundled with Node LTS |

**Docker Build additional prerequisite:**

| Prerequisite | Version | Notes |
| ------------ | ------- | ----- |
| [Docker CE](https://docs.docker.com/install/) | `Stable` | - |

> 

Author: Thank you - glad you added it to this PR rather than to a new PR against the Docker setup doc!

> 

Reviewer 2: No problem!  I'm glad I checked first...
